src/main/java/org/cru/redegg/recording/log4j2/Log4j2Recorder.java

```diff
@@ -0,0 +1,50 @@
+package org.cru.redegg.recording.log4j2;
+
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.core.LoggerContext;
+import org.apache.logging.log4j.core.config.AbstractConfiguration;
+import org.apache.logging.log4j.core.config.Configuration;
+import org.apache.logging.log4j.core.config.LoggerConfig;
+import org.cru.redegg.recording.api.LoggingRecorder;
+import org.cru.redegg.recording.api.RecorderFactory;
+
+import java.util.Set;
+
+/**
+ * Encapsulates a dependency on log4j2.
+ *
+ * @author Matt Drees
+ */
+public class Log4j2Recorder implements LoggingRecorder
+{
+    private RedEggLog4j2Appender appender;
+    private final LoggerContext loggerContext;
+
+    public static Log4j2Recorder add(RecorderFactory recorderFactory, Set<String> ignoredLoggers)
+    {
+        return new Log4j2Recorder(recorderFactory, ignoredLoggers);
+    }
+
+    private Log4j2Recorder(RecorderFactory recorderFactory, Set<String> ignoredLoggers)
+    {
+        appender = new RedEggLog4j2Appender(recorderFactory, ignoredLoggers);
+        LogManager.getRootLogger().info("adding log4j appender");
+        loggerContext = (LoggerContext) LogManager.getContext(false);
+        final Configuration config = loggerContext.getConfiguration();
+        appender.start();
+        config.addAppender(appender);
+        LoggerConfig loggerConfig = config.getRootLogger();
```

Comment:
This only applies to the root logger, should this also get applied to the other loggers that may be defined by using: `config.getLoggers().values().forEach(c -> c.addAppender(appender, Level.DEBUG, null));`?

I'll run a quick test to see what happens.

Reply:
Ah, I guess if you use additivity=false that might be helpful. I've not used that much, so I didn't think about that.

Hmm.

Reply:
ah, good to know I can adjust it with additivity, I really need to work on the CAS logging config at some point. Leave it just the root logger for now.

Reply:
for authentication (bad password?) messages that get logged as ERROR, we should suppress those with ErrorRecorder.ignoreErrorsFromLogger()

Reply:
perfect, that's probably what I was looking for but not finding :)

Reply:
yeah, it probably doesn't make sense for it to live in ErrorRecorder. Sorry about that.
(I had wanted to be able to suppress differently depending on the request, since I wrote that originally for an oracle error logger that I only wanted to suppress in a weird corner case, so I didn't make it a global config option)

Reply:
yeah, I'm finding it's hard to get access to that knob from my code...

Reply:
this whole conversation can be revisited when I get around to adjusting the logging configuration for the key

Reply:
ok, sounds good